src/gui/editors/PianoRoll.cpp

```diff
@@ -627,6 +628,72 @@ void PianoRoll::clearGhostPattern()
 }
 
 
+void PianoRoll::glueNotes()
+{
+	if (hasValidPattern())
+	{
+		NoteVector selectedNotes = getSelectedNotes();
+		if (selectedNotes.empty())
+		{
+			TextFloat::displayMessage( tr( "Glue notes failed" ),
+					tr( "Please select notes to glue first." ),
+					embed::getIconPixmap( "glue", 24, 24 ),
+					3000 );
+			return;
+		}
+
+		// Make undo possible
+		m_pattern->addJournalCheckPoint();
+
+		// Sort notes on key and then pos.
+		std::sort(selectedNotes.begin(), selectedNotes.end(),
+			[](const Note * note, const Note * compareNote) -> bool
+			{
+				if (note->key() == compareNote->key())
+				{
+					return note->pos() < compareNote->pos();
+				}
+				return note->key() < compareNote->key();
+			});
+
+		QList<Note *> noteToRemove;
+
+		NoteVector::iterator note = selectedNotes.begin();
+		auto nextNote = note+1;
+		NoteVector::iterator end = selectedNotes.end();
+
+		while (note != end && nextNote != end)
+		{
+			// key and position match for glue
+			if ((*note)->key() == (*nextNote)->key()
+				&& (*nextNote)->pos() >= (*note)->pos()
```

Comment:
Will that conditional ever be `false`? Since the notes were previously ordered by key and by pos, if two notes have the same key isn't it always `true` that the next note will have a position equal or greater than the previous note?

Reply:
The comment highlighted all 4 lines, so to make it clearer the conditional I meant is `(*nextNote)->pos() >= (*note)->pos()` which will be evaluated if the `key`s are the same on the two notes.

Reply:
It is redundant, but I think there should be a comment which explains what you said.

Reply:
So, what needs to be done here? Remove `&& (*nextNote)->pos() >= (*note)->pos()` and improve the comment?

Reply:
Yeah, I think just removing that part of the conditional and having the comment explaining that you don't need to check if the next note is after the first because they are ordered already. Removes an unnecessary comparison but leaves an explanation why.

Reply:
OK. Hopefully fixed in https://github.com/LMMS/lmms/pull/5482/commits/d86ba8c17d461922e082b5a93c894b6c63506627